source/common/http/conn_manager_impl.cc

```diff
@@ -484,6 +504,25 @@ void ConnectionManagerImpl::doConnectionClose(
   }
 }
 
+void ConnectionManagerImpl::createStartDrainTimer(std::chrono::milliseconds drain_delay) {
+  start_drain_timer_ = read_callbacks_->connection().dispatcher().createTimer([this]() -> void {
+    if (drain_state_ != DrainState::NotDraining) {
+      return;
+    }
+    if (!codec_) {
+      stats_.named_.downstream_cx_drain_close_.inc();
+      doConnectionClose(Network::ConnectionCloseType::FlushWrite, absl::nullopt, "");
+    } else {
+      startDrainSequence();
+      stats_.named_.downstream_cx_drain_close_.inc();
+      for (const auto& stream : streams_) {
+        ENVOY_STREAM_LOG(debug, "drain closing connection", *stream);
+      }
```

Comment:
I'm not sure why this logging is here instead of startDrainSequence.  What makes the start of a drain timer different from other calls to startDrainSequence?